pints/_abc/_abc_smc.py

```diff
@@ -0,0 +1,198 @@
+#
+# ABC SMC method
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import pints
+import numpy as np
+
+
+class ABCSMC(pints.ABCSampler):
+    r"""
+    Implements the ABC-SMC algorithm as describd in [1].
+
+    In each iteration of the algorithm, the following steps occur:
+
+    .. math::
+        \begin{align}
+        & \theta^* \sim p_{t-1}(\theta) \textrm{, i.e. sample parameters from
+        previous intermediate distribution} \\
+        & \theta^{**} \sim K(\theta|\theta^{*}), \textrm{i.e. perturb }
+        \theta^{*} \textrm{ to     obtain to new point } x \sim
+        p(x|\theta^{**})\textrm{, i.e. sample data from sampling
+        distribution} \\
+        & \textrm{if } s(x) < \textrm{threshold}_(t), \theta^* \textrm{
+        added to list of samples[t]}
+        \end{align}
+
+    After we have obtained nr_samples samples, t is advanced, and weights
+    are calculated for samples[t-1]. At the last error threshold,
+    samples are returned whenever they are accepted. This algorithm is
+    also referred to as ABC Population Monte Carlo (ABC PMC) [2].
+
+    References
+    ----------
+    .. [1] "Toni, Tina, et al. Approximate Bayesian computation scheme
+            for parameter inference and model selection in dynamical systems.
+            Journal of the Royal Society Interface, 6.31: 187-202, 2009.
+            https://doi.org/10.1098/rsif.2008.0172
+
+    .. [2] "Beaumont, Mark A., et al. Adaptive approximate Bayesian
+            computation. Biometrika, 96.4: 983-990, 2009."
+            https://doi.org/10.48550/arXiv.0805.2256
+
+    Parameters
+    ----------
+    nr_samples
+        The number of samples requested for intermediate distributions.
+    error_schedule
+        The schedule of error threshold distance for all distributions.
+    """
+
+    def __init__(self, log_prior, perturbation_kernel=None):
```

Comment:
Why is the kernel a constructor arg (while the schedule is not)?
Should this be changeable after construction? (like the schedule again)